MediaBrowser.MediaEncoding/Encoder/MediaEncoder.cs

```diff
@@ -772,6 +775,195 @@ private async Task<string> ExtractImageInternal(string inputPath, string contain
+            // Additionally, we must set a few fields without defaults to prevent null pointer exceptions.
+            if (!allowHwAccel)
+            {
+                options.EnableHardwareEncoding = false;
+                options.HardwareAccelerationType = string.Empty;
+                options.EnableTonemapping = false;
+            }
+
+            var baseRequest = new BaseEncodingJobOptions { MaxWidth = maxWidth };
+            var jobState = new EncodingJobInfo(TranscodingJobType.Progressive)
+            {
+                IsVideoRequest = true,  // must be true for InputVideoHwaccelArgs to return non-empty value
+                MediaSource = mediaSource,
+                VideoStream = imageStream,
+                BaseRequest = baseRequest,  // GetVideoProcessingFilterParam errors if null
+                MediaPath = inputFile,
+                OutputVideoCodec = "mjpeg"
+            };
+            var vidEncoder = options.AllowMjpegEncoding ? encodingHelper.GetVideoEncoder(jobState, options) : jobState.OutputVideoCodec;
+
+            // Get input and filter arguments
+            var inputArg = encodingHelper.GetInputArgument(jobState, options, container).Trim();
+            if (string.IsNullOrWhiteSpace(inputArg))
+            {
+                throw new InvalidOperationException("EncodingHelper returned empty input arguments.");
+            }
+
+            if (!allowHwAccel)
+            {
+                inputArg = "-threads " + threads + " " + inputArg; // HW accel args set a different input thread count, only set if disabled
+            }
+
+            var filterParam = encodingHelper.GetVideoProcessingFilterParam(jobState, options, jobState.OutputVideoCodec).Trim();
+            if (string.IsNullOrWhiteSpace(filterParam) || filterParam.IndexOf("\"", StringComparison.Ordinal) == -1)
+            {
+                throw new InvalidOperationException("EncodingHelper returned empty or invalid filter parameters.");
+            }
+
+            return ExtractVideoImagesOnIntervalInternal(inputArg, filterParam, interval, vidEncoder, threads, qualityScale, priority, cancellationToken);
+        }
+
+        private async Task<string> ExtractVideoImagesOnIntervalInternal(
+            string inputArg,
+            string filterParam,
+            TimeSpan interval,
+            string vidEncoder,
+            int? outputThreads,
+            int? qualityScale,
+            ProcessPriorityClass? priority,
+            CancellationToken cancellationToken)
+        {
+            if (string.IsNullOrWhiteSpace(inputArg))
+            {
+                throw new InvalidOperationException("Empty or invalid input argument.");
+            }
+
+            // Output arguments
+            string fps = "fps=1/" + interval.TotalSeconds.ToString(CultureInfo.InvariantCulture);
+            if (string.IsNullOrWhiteSpace(filterParam))
+            {
+                filterParam = "-vf \"" + fps + "\"";
```

Comment:
Better to move the `fps` filter to `GetVideoProcessingFilterParam()` and set the fps value in the request.
We may need reuse this filter to to limit the video frame rate.

Reply:
I've made the change, but realized there is a possibility for both the ```-r``` option and ```fps``` option to be set when ```GetVideoQualityParam``` is used in conjunction with ```GetVideoProcessingFilterParam```. I'm not entirely sure if the simultaneous use of these options would cause ```-r``` to further lower the rate set by the fps filter, or if one would take precedence over the other. 

As far as code consistency and ffmpeg functionality, what are your thoughts? Would removing the  ```-r``` option from ```GetVideoQualityParam``` and leaving it solely up to ```GetVideoProcessingFilterParam``` lead to any issues? It seems like ```GetVideoQualityParam``` is always used after the input argument so I imagine fps should work to lower the output video fps as well.